department-of-veterans-affairs/va.gov-cms
# [draft] VACMS-8345 design system

## Description
Closes #8345,  #8346. 

this PR does quite a bit!

- Removes very old legacy stuff from the design-system folder. Like, stuff from the earliest ages of this project. We are using it for our design system now.

- It brings in Storybook 6 with support for SCSS files & Drupal-specific Twig (think `attributes`, `|clean_id`, `|t` etc.). Not every Twig function behaves exactly how drupal does (`|render` etc), but they won't break our components in SB and allow our templates to work in both places easily. 

- Probably some other storybook addons we will want to add/configure as well (a11y, viewports, etc.)

- Small helper script for getting CSS variables into a usable format for display in storybook (dumps them into a generated .json file)

- Stubbed out the alert & block component as an example. Also started some basic structuring for overall layout of stories. There is some duplication of scss & templates from the theme right now... will be fixed before this PR is off draft.

- Adds command to bundle design system assets, sans-storybook file, for Drupal consumption with appropriate namespacing for use in vagovclaro.

- Adds composer commands to watch (dev) & build (prod) storybook, & building for vagovclaro

## Testing done
- Verified it runs locally and that SB controls can update templates without refreshing
- Verified that compiling static pages (for hosting in tugboat) work without error
- Verified that no new security alerts will be added to the repo as part of these packages

```
 heffner@MacBook-Pro  ~/dev/va.gov-cms/docroot/design-system   VACMS-8187-design-system ±✚  yar
n audit
yarn audit v1.22.17
0 vulnerabilities found - Packages audited: 1349
✨  Done in 0.89s.
```

## Screenshots
layout of stories:
<img width="233" alt="Screen Shot 2022-03-16 at 3 38 49 PM" src="https://user-images.githubusercontent.com/11279744/158703522-60886f9c-2361-40f3-95e5-138385261646.png">

alert stub:
<img width="946" alt="Screen Shot 2022-03-16 at 3 25 38 PM" src="https://user-images.githubusercontent.com/11279744/158703552-dda0ba66-984b-43f5-8650-1b321d32ebea.png">

controls connected:
<img width="1668" alt="Screen Shot 2022-03-16 at 3 39 00 PM" src="https://user-images.githubusercontent.com/11279744/158703535-af140cdb-cc9a-483c-af7b-2d4268fe8ffd.png">
<img width="1689" alt="Screen Shot 2022-03-16 at 3 39 13 PM" src="https://user-images.githubusercontent.com/11279744/158703536-3d220069-0880-437b-9760-beaee2b270d7.png">


## QA steps
Checkout branch locally... you may need to do this from my fork?

```sh
cd docroot/design-system
yarn install
yarn run build-storybook # to verify static files build without error
yarn run storybook # to see running locally in your browser
```

Click around the running instance and see that controls behave for the most part (drupalAttribute is iffy, we probably hide that control in the future and only worry about text boxes etc.)

### Definition of Done


### Select Team for PR review


### CMS user-facing annoucement
